packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1173,46 +988,52 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
-    function _priceRequestResolved(
-        PriceRequest storage priceRequest,
-        VoteInstance storage voteInstance,
-        uint256 currentRoundId
-    ) private returns (bool) {
-        // We are currently either in the voting round for the request or voting is yet to begin.
-        if (currentRoundId <= priceRequest.lastVotingRound) return false;
-
-        // If the request has been previously resolved, return true.
-        if (priceRequest.pendingRequestIndex == UINT64_MAX) return true;
-
-        // Else, check if the price can be resolved.
-        (bool isResolvable, int256 resolvedPrice) =
-            voteInstance.resultComputation.getResolvedPrice(_computeGat(priceRequest.lastVotingRound));
-
-        // If it's not resolvable return false.
-        if (!isResolvable) return false;
-
-        // Else, the request is resolvable. Remove the element from the pending requests and update pendingRequestIndex
-        // within the price request struct to make the next entry into this method a no-op for this request.
-        _removeRequestFromPendingPriceRequests(priceRequest.pendingRequestIndex);
-
-        priceRequest.pendingRequestIndex = UINT64_MAX;
-        emit PriceResolved(
-            priceRequest.lastVotingRound,
-            priceRequest.priceRequestIndex,
-            priceRequest.identifier,
-            priceRequest.time,
-            priceRequest.ancillaryData,
-            resolvedPrice
-        );
-        return true;
+    function _resolveResolvablePriceRequests() private {
+        uint32 currentRoundId = uint32(getCurrentRoundId());
+
+        uint256 index = 0;
+        while (index < pendingPriceRequestsIds.length) {
+            PriceRequest storage request = priceRequests[pendingPriceRequestsIds[index]];
+            VoteInstance storage voteInstance = request.voteInstances[request.lastVotingRound];
+            (bool isResolvable, int256 resolvedPrice) =
+                voteInstance.resultComputation.getResolvedPrice(_computeGat(request.lastVotingRound));
+            if (!isResolvable) {
+                if (request.lastVotingRound < currentRoundId) {
+                    request.rollCount++;
+                    request.lastVotingRound = currentRoundId;
+                    if (request.rollCount == deleteAfterRollCount && !request.isGovernance) {
+                        emit RequestDeleted(request.identifier, request.time, request.ancillaryData);
+
+                        delete priceRequests[pendingPriceRequestsIds[index]];
+                        _removeRequestFromPendingPriceRequestsIds(request.pendingRequestIndex);
+                    }
+                } else index++;
+                continue;
+            }
+
+            request.resolvedRequestIndex = SafeCast.toUint64(resolvedPriceRequestIds.length);
+            resolvedPriceRequestIds.push(_encodePriceRequest(request.identifier, request.time, request.ancillaryData));
```

Comment:
nit: why not just read this value from the `pendingPriceRequestsIds[index]` instead of re-encoding?

Reply:
agreed. done.